### PR TITLE
Pte 13 2 synchronizate game

### PR DESCRIPTION
**Pull request by Hidennnn**
Now game is fully handle by server. Players can quit, collide with objects. Players can win by meta

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -2,10 +2,60 @@
...
+
+FRAME_TIME = 1 / 60
+
+GAME_IS_STARTED = True
+
+PLAYER_POSITION_Y = 200
+
+STARTING_Y = 200
+
+STARTING_X = 100
```
Comment (Misiek22113) [inline server.py]: reduce this empty lines

File: server.py [hunk 2]
```diff
@@ -84,9 +138,18 @@ def __init__(self):
 
         # variables for game
         self.chosen_champions = [CHAMPION_IS_NOT_CHOSEN for _ in range(MAX_PLAYERS)]
+        self.players = [None, None, None]
```
Comment (Misiek22113) [inline server.py]: add some constans like EMPTY_SLOT and replace this None

File: server.py [hunk 3]
```diff
@@ -141,68 +204,211 @@ def connect_client(self):
...
-            return LEFT_CHAMPION_SELECT
+            if self.server_status == SERVER_IS_CLOSING:
+                self.client_sockets[client_number].sendall(SERVER_QUIT_SIGNAL_IN_CHAMPION_SELECT)
+                return LEFT_CHAMPION_SELECT
 
-        if chosen_champion in self.chosen_champions:
-            self.client_sockets[client_number].sendall(CHAMPION_IS_NOT_AVAILABLE)
-            return CHAMPION_SELECT
+            if chosen_champion in self.chosen_champions:
+                self.client_sockets[client_number].sendall(CHAMPION_IS_NOT_AVAILABLE)
```
Comment (Misiek22113) [inline server.py:232]: aren't those if statements to deep? Like to far away from left border of the file?

Comment (Hidennnn) [reply server.py:232]: due to try

File: server.py [hunk 4]
```diff
@@ -141,68 +204,211 @@ def connect_client(self):
...
-            self.client_sockets[client_number].send(pickle.dumps(SERVER_QUIT_SIGNAL_IN_LOBBY))
+            if self.server_status == SERVER_IS_CLOSING:
+                self.client_sockets[client_number].send(pickle.dumps(SERVER_QUIT_SIGNAL_IN_LOBBY))
+                self.client_sockets[client_number].send(pickle.dumps(SERVER_QUIT_SIGNAL_IN_LOBBY))
+                return CONNECTION_IS_DEACTIVATED
+
+            if client_signal == CHOSEN_CHAMPIONS_INFORMATION_REQUEST:
+                self.client_sockets[client_number].send(pickle.dumps(self.chosen_champions))
+
+                count = EVERY_PLAYER_CONNECTED
```
Comment (Misiek22113) [inline server.py]: instead of count I would add active_players or players_in_lobby

File: server.py (hunk 1 above)
Comment (Hidennnn) [reply server.py]: changed + better grouped

File: server.py (hunk 2 above)
Comment (Hidennnn) [reply server.py]: replaced

File: server.py (hunk 4 above)
Comment (Hidennnn) [reply server.py]: changed

File: src/menu/lobby/lobby.py [hunk 5]
```diff
@@ -6,6 +6,16 @@
...
+PLAYER_Y = 350
+
+PLAYER_X = 240
+
+CHAMPION_NAME = 1
+
+GAME_IS_STARTED = 1
+
+BUFFER_SIZE = 4096
+
```
Comment (Misiek22113) [inline src/menu/lobby/lobby.py]: useless empty lines

**Review by Misiek22113 — APPROVED** [review]

File: src/menu/lobby/lobby.py (hunk 5 above)
Comment (Hidennnn) [reply src/menu/lobby/lobby.py]: done

**Pull request merged**